crates/environ/src/module.rs

```diff
@@ -428,7 +428,7 @@ impl ModuleTranslation<'_> {
 
             // If this is not a funcref table, then we can't support a
             // pre-computed table of function indices.
-            if self.module.table_plans[segment.table_index].table.wasm_ty != WasmType::FuncRef {
+            if self.module.table_plans[segment.table_index].table.wasm_ty.heap_type == WasmHeapType::Func {
                 leftovers.push(segment.clone());
                 continue;
             }
```

Comment:
I think that this condition isn't quite right, you'll probably want to `match` on the `heap_type` here and explicitly reject `externref` but both typed functions and normal functions should be able to fall through.

Reply:
It now reads:
```rust
if self.module.table_plans[segment.table_index]
                .table
                .wasm_ty
                .heap_type
                == WasmHeapType::Extern
            {
                leftovers.push(segment.clone());
                continue;
            }
```

Is this close enough to what you had in mind? Alternatively, we can match explicitly on `heap_type` to make this code more robust against changes to `WasmHeapType`.